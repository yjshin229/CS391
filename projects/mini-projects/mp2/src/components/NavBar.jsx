import { styled } from "styled-components";

const NavBar = ({ title }) => {
  return (
    <StyledNav>
      <StyledHeader>{title}</StyledHeader>
    </StyledNav>
  );
};
export default NavBar;

const StyledNav = styled.nav`
  background-color: #df8321;
  padding: 0rem 2rem;
`;

const StyledHeader = styled.h1`
  color: white;
  font-family: "Protest Guerrilla", sans-serif;
  font-size: 42px;
`;
