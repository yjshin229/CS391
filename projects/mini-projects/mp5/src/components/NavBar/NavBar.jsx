import styled from "styled-components";
import Button from "../Button/Button";
import PropTypes from "prop-types";

const NavBar = ({ goBack, title, openForm = false }) => {
  return (
    <NavBarWrapper>
      {goBack && <NavBarButton onClick={goBack}>{`< Go Back`}</NavBarButton>}
      <Title>{title}</Title>
      {openForm && (
        <NavBarButton
          onClick={openForm}
          data-cy="addReview"
        >{`+ Add Review`}</NavBarButton>
      )}
    </NavBarWrapper>
  );
};

export default NavBar;

NavBar.propTypes = {
  goBack: PropTypes.func,
  title: PropTypes.string,
  openForm: PropTypes.bool,
};

const NavBarWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  background: cornflowerBlue;
`;

const Title = styled.h2`
  text-align: center;
  flex-basis: 60%;
  &:first-child {
    margin-left: 20%;
  }
  &:last-child {
    margin-right: 20%;
  }
`;

const NavBarButton = styled(Button)`
  margin: 10px 5%;
`;
