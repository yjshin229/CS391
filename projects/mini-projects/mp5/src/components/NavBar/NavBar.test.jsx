import { render, screen, fireEvent } from "@testing-library/react";
import { test, describe, expect, vi, renderer } from "vitest";
import NavBar from "./NavBar";

describe("Rendering test", () => {
  test("The NavBar component should render", () => {
    const view = renderer
      .create(<NavBar />)
      .toJSON()
      .expect(view)
      .toMatchSnapshot();
  });

  test("The NavBar component should render with a title", () => {
    const title = "Test application";
    render(<NavBar title={title} />);

    expect(screen.getByRole("heading")).toHaveTextContent(title);
  });
});

test("The NavBar component should respond to button clicks", () => {
  const mockFunction = vi.fn();
  render(<NavBar goBack={mockFunction} openForm={mockFunction} />);

  fireEvent.click(screen.getByText("< Go Back"));
  expect(mockFunction).toHaveBeenCalled();

  fireEvent.click(screen.getByText("+ Add Review"));
  expect(mockFunction).toHaveBeenCalledTimes(2);
});
