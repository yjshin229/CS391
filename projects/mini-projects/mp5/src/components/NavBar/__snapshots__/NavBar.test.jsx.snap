// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rendering test > The NavBar component should render 1`] = `
<div
  className="sc-guDLey gtpvTe"
>
  <h2
    className="sc-dmyCSP jbzSYl"
  />
</div>
`;

exports[`The NavBar component should render 1`] = `
<div
  className="sc-guDLey gtpvTe"
>
  <h2
    className="sc-dmyCSP jbzSYl"
  />
</div>
`;
